akka-actor/src/main/scala/akka/actor/Address.scala

```diff
@@ -31,6 +32,16 @@ final case class Address private (protocol: String, system: String, host: Option
   def this(protocol: String, system: String) = this(protocol, system, None, None)
   def this(protocol: String, system: String, host: String, port: Int) = this(protocol, system, Option(host), Some(port))
 
+  /**
+   * Java API: The hostname if specified or empty optional if not
+   */
+  def getHost(): Optional[String] = host.asJava
+
+  /**
+   * Java API: The port if specified or empty optional if not
+   */
+  def getPort(): Optional[Int] = port.asJava
```

Comment:
This compiles to a `public java.util.Optional<java.lang.Object> getPort();`. Not really sure how to write that nicely though...
```suggestion
  def getPort(): Optional[Int] = port.map[Integer](i => i)asJava
```

Reply:
Good catch, I'll see about it.